### PR TITLE
udhcpc [option 42] configuration for ntp

### PR DESCRIPTION
**Pull request by devinslick**
For reference: https://github.com/armcc/udhcp/blob/master/README.udhcpc

### CODE REVIEW

File: overlay/lower/usr/share/udhcpc/default.script [hunk 1]
```diff
@@ -73,6 +73,17 @@ case "$1" in
...
 			echo "nameserver $i" >>$tmpfile
 		done
+                if [ -n "$ntpsrv" ]; then
+                        echo "NTP servers provided by DHCP: $ntpsrv"
+                        # Erase /etc/ntp.conf before adding new servers
+                        > /etc/ntp.conf
+                        # Add NTP servers to /etc/ntp.conf
+                        for t in $ntpsrv; do
+                                echo "Adding NTP server $t"
+                                echo "server $t iburst" >> /etc/ntp.conf
```
Comment (themactep) [inline overlay/lower/usr/share/udhcpc/default.script:83]: this is an issue. constant writing to the flash on reconnects will degrade flash chip.

Comment (gtxaspec) [reply overlay/lower/usr/share/udhcpc/default.script:83]: Should we move it to tmp? Symlink?

Comment (themactep) [reply overlay/lower/usr/share/udhcpc/default.script:83]: Yes. We could copy the cached version to tmp and make it a working copy if nothing is sent in dhcp, or override the working copy with servers from dhcp.

**Comment by devinslick** [comment]
@themactep / @gtxaspec ,
Like this?

`
                if [ -n "$ntpsrv" ]; then
                        echo $ntpsrv | tr ' ' '\n' | sed 's/^/server /; s/$/ iburst/' > /tmp/ntp.conf
                        rm /etc/ntp.conf
                        ln -s /tmp/ntp.conf /etc/ntp.conf
                        /etc/init.d/S49ntpd reload
                fi
`
If so, any concerns with the removal of /etc/ntp.conf?  Does it need to be backed-up and conditionally restored?

**Comment by gtxaspec** [comment]
probably like:

`/etc/ntpd.conf` on the overlay is a permanent symlink to `/tmp/ntpd.conf`

The dhcp script would over-ride the default set of servers if provided by the network,

`S49ntpd` in `/etc/init.d/` creates `/tmp/ntpd.conf` with a default set of servers if none were provided by the network.


Sound right?

**Comment by themactep** [comment]
Sounds legit. Let's try that.

**Comment by devinslick** [comment]
Alright @gtxaspec / @themactep , WDYT?

**Comment by themactep** [comment]
Looks very good. Please correct the couple of hiccups, then i'll make a fresh build to test locally, and if nothing fails in staging, i'll merge it. Thank you.

**Pull request merged**